scripts/mirror.js

```diff
@@ -0,0 +1,609 @@
+        : [true, false, null, undefined];
+    if (Array.isArray(comp[browser])) {
+      for (let i = 0; i < comp[browser].length; i++) {
+        if (triggers.includes(comp[browser][i].version_added)) {
+          doBump = true;
+          break;
+        }
+      }
+    } else if (comp[browser] !== undefined) {
+      doBump = triggers.includes(comp[browser].version_added);
+    } else {
+      doBump = true;
+    }
+  }
+
+  if (doBump) {
+    let newValue = bumpVersion(comp[source], browser, source, comp[browser]);
+    if (newValue !== null) {
+      newData[rootPath].__compat.support[browser] = newValue;
+    }
+  }
+
+  return newData;
+};
+
+/**
+ * @param {Identifier} data
+ * @param {string} feature
+ * @param {string} browser
+ * @param {string} source
+ * @param {string} modify
+ * @returns {Identifier}
+ */
+const setFeature = (data, feature, browser, source, modify) => {
+  let newData = Object.assign({}, data);
+
+  const rootPath = feature.shift();
+  if (feature.length > 0 && data[rootPath].constructor == Object) {
+    newData[rootPath] = setFeature(
+      data[rootPath],
+      feature,
+      browser,
+      source,
+      modify,
+    );
+  } else {
+    if (data[rootPath].constructor == Object || Array.isArray(data[rootPath])) {
+      newData = doSetFeature(data, newData, rootPath, browser, source, modify);
+    }
+  }
+
+  return newData;
+};
+
+/**
+ * @param {Identifier} data
+ * @param {string} browser
+ * @param {string} source
+ * @param {string} modify
+ * @returns {Identifier}
+ */
+const setFeatureRecursive = (data, browser, source, modify) => {
+  let newData = Object.assign({}, data);
+
+  for (let i in data) {
+    if (!!data[i] && typeof data[i] == 'object' && i !== '__compat') {
+      newData[i] = data[i];
+      if (data[i].__compat) {
+        doSetFeature(data, newData, i, browser, source, modify);
+      }
+      setFeatureRecursive(data[i], browser, source, modify);
+    }
+  }
+
+  return newData;
+};
+
+/**
+ * @param {string} browser
+ * @param {string} filepath
+ * @param {string} source
+ * @param {string} modify
+ * @returns {boolean}
+ */
+function mirrorDataByFile(browser, filepath, source, modify) {
+  let file = filepath;
+  if (file.indexOf(__dirname) !== 0) {
+    file = path.resolve(__dirname, '..', file);
+  }
+
+  if (!fs.existsSync(file)) {
+    return false;
+  }
+
+  if (fs.statSync(file).isFile()) {
+    if (path.extname(file) === '.json') {
+      let data = require(file);
+      let newData = setFeatureRecursive(data, browser, source, modify);
+
+      fs.writeFileSync(file, JSON.stringify(newData, null, 2) + '\n', 'utf-8');
+    }
+  } else if (fs.statSync(file).isDirectory()) {
+    const subFiles = fs.readdirSync(file).map(subfile => {
+      return path.join(file, subfile);
+    });
+
+    for (let subfile of subFiles) {
+      mirrorDataByFile(browser, subfile, source, modify);
+    }
+  }
+
+  return true;
+}
+
+/**
+ * @param {string} browser
```

Comment:
I'm not sure if we should really support `mirrorDataByFeature`, but if we do, we need to warn about its brokenness.

```suggestion
/**
 * Allows mirroring by feature ID (e.g. "html.elements.a")
 *
 * Note that this assumes a predictable file structure
 * which BCD doesn't have right now. (issue #3617)
 * For example, even if "html.elements.input.input-button"
 * is a valid query, it will fail here.
 *
 * @param {string} browser
```

Reply:
I find myself using mirroring data by feature identifiers a **lot** more than by filename, so I am personally leaning towards keeping it and eventually fixing it so that it'll work regardless of the file structure (or if #3617 is fixed)!